frontend/src/components/Navigator/index.js

```diff
@@ -215,11 +221,13 @@ function Navigator({
     // Prevent user from making changes to goal title during auto-save.
     setSavingLoadScreen(isAutoSave);
 
+    const endDate = formEndDate && formEndDate.toLowerCase() !== 'invalid date' ? formEndDate : '';
```

Comment:
instead of relying on this exact string, the following might work (untested):

```javascript
moment(formEndDate).isValid()
```

https://momentjscom.readthedocs.io/en/latest/moment/01-parsing/15-is-valid/